src/components/include/DefaultImportModalDialog.js

```diff
@@ -0,0 +1,177 @@
+import React from 'react';
+import _ from 'underscore';
+import {Made} from "@exabyte-io/made.js";
+import {ModalHeader, ModalBody} from 'react-bootstrap';
+
+
+import {Material} from "../../material";
+import {ModalDialog} from './ModalDialog';
+import BasisText from "../source_editor/BasisText";
+
+
+export const defaultXYZMaterialConfig = {
+    name: 'Custom XYZ',
+    basis: {
+        elements: [
+            {
+                id: 1,
+                value: 'H'
+            }
+        ],
+        coordinates: [
+            {
+                id: 1,
+                value: [
+                    0.00,
+                    0.00,
+                    0.00
+                ]
+            }
+        ],
+        units: Made.ATOMIC_COORD_UNITS.cartesian
+    },
+    lattice: {
+        type: Made.LATTICE_TYPE_CONFIGS[1].code, // FCC, to be exported by made.js
+        a: 500.0, // NB will be rewritten with mockLatticeConstant
+        b: 500.0,
+        c: 500.0,
+        alpha: 90,
+        beta: 90,
+        gamma: 90,
+        units: {
+            length: Made.units.angstrom,
+            angle: Made.units.degree
+        }
+    }
+};
+
+const mockLatticeConstantMultiplicator = 2.0;
+
+class DefaultImportModalDialog extends ModalDialog {
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            xyz: "Paste your XYZ"
+        };
+        this.handleSubmit = this.handleSubmit.bind(this);
+        this.handleChange = this.handleChange.bind(this);
+        this.handleFileRead = this.handleFileRead.bind(this);
+
+        this.reader = new FileReader();
+        this.reader.onloadend = this.handleFileRead;
+    }
+
+    handleFileRead(evt) {
+        this.setState({xyz: evt.target.result});
+    }
+
+    handleSubmit() {
+
+        if (!this.BasisTextComponent.state.isContentValidated) return;
+
+        const basisConfig = Made.parsers.xyz.toBasisConfig(this.state.xyz, Made.ATOMIC_COORD_UNITS.cartesian);
+
+        // determine the max distance to use for cell for user's XYZ
+        const x = [],
+            y = [],
+            z = [];
+        basisConfig.coordinates.forEach(function(item){
+            x.push(item.value[0]);
+            y.push(item.value[1]);
+            z.push(item.value[2]);
+        });
+
+        const diffX = _.max(x) - _.min(x),
+            diffY = _.max(y) - _.min(y),
+            diffZ = _.max(z) - _.min(z);
+        const mockLatticeConstant = _.max([diffX, diffY, diffZ]) * mockLatticeConstantMultiplicator;
```

Comment:
Lines 75-88 should go elsewhere. We should probably move this to Made, maybe to `utils` or `Basis` there.